Chessington.GameEngine/Helpers.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Chessington.GameEngine
+{
+	public class Helpers
+    {
+        public List<Direction> AddMovesAndUpdateDirections(Board board, List<Direction> directions, Square currentSquare, int i, List<Square> moves, Player player)
+        {
+            var toRemove = new List<Direction>();
+            foreach (var direction in directions)
+            {
+                var potentialMove = Square.At(currentSquare.Row + i * direction.RowOffset,
+                    currentSquare.Col + i * direction.ColOffset);
+                if (board.GetPiece(potentialMove) != null)
+                {
+                    toRemove.Add(direction);
+                    if (board.GetPiece(potentialMove).Player != player)
+                    {
+                        moves.AddIfOnBoard(potentialMove);
+                    }
+                }
+                else
+                {
+                    moves.AddIfOnBoard(potentialMove);
+                }
+            }
+
+            return directions.Except(toRemove).ToList();
+        }
```

Comment:
This method has a fair amount of nesting. It also both updates the moves list and returns a new directions list, which is quite unexpected. A method should generally either mutate it's input or return new output, not both (and I prefer returning new output, but that's personal preference). I would suggest splitting this into two functions; one for updating the directions and one for updating the moves. It would mean looping through the directions twice, but I think it's worth it for more readable code 🙂